packages/angular_devkit/build_facade/src/architect.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  BaseException,
+  JsonObject,
+  Path,
+  getSystemPath,
+  logging,
+  normalize,
+  resolve,
+  virtualFs,
+} from '@angular-devkit/core';
+import { NodeJsSyncHost, createConsoleLogger } from '@angular-devkit/core/node';
+import { resolve as nodeResolve } from '@angular-devkit/core/node';
+import * as fs from 'fs';
+import { Observable } from 'rxjs/Observable';
+import { BuildFacade } from './build-facade';
+import { BuildEvent, Target } from './target';
+import { Workspace } from './workspace';
+
+
+export class ProjectNotFoundException extends BaseException {
+  constructor(name?: string) {
+    const nameOrDefault = name ? `Project '${name}'` : `Default project`;
+    super(`${nameOrDefault} could not be found in workspace.`);
+  }
+}
+
+export class TargetNotFoundException extends BaseException {
+  constructor(name?: string) {
+    const nameOrDefault = name ? `Target '${name}'` : `Default target`;
+    super(`${nameOrDefault} could not be found in workspace.`);
+  }
+}
+
+export class ConfigurationNotFoundException extends BaseException {
+  constructor(name: string) {
+    super(`Configuration '${name}' could not be found in project.`);
+  }
+}
+
+export class TargetNotSupportedException extends BaseException {
+  constructor(targetProperties: TargetProperties) {
+    super(`Build facade '${targetProperties.buildFacade}' does not support` +
+      `'${targetProperties.targetType}' target.`);
+  }
+}
+
+export class InvalidBuilderException extends BaseException {
+  constructor(targetProperties: TargetProperties) {
+    super(`Build facade '${targetProperties.buildFacade}' did not return a valid builder ` +
+      `for the following options: ${JSON.stringify(targetProperties, null, 2)}`);
+  }
+}
+
+export class InvalidRunnerException extends BaseException {
+  constructor(targetProperties: TargetProperties) {
+    super(`Build facade '${targetProperties.buildFacade}' did not return a valid runner ` +
+      `for the following options: ${JSON.stringify(targetProperties, null, 2)}`);
+  }
+}
+
+export interface TargetProperties {
+  root: Path;
+  projectType: string;
+  buildFacade: string;
+  targetType: string;
+}
+
+export interface TargetInfo<OptionsT = JsonObject> {
+  properties: TargetProperties;
+  options: OptionsT;
+}
+
+export class Architect {
+  _host: virtualFs.SyncDelegateHost<fs.Stats>;
```

Comment:
Does it need to be `Sync`?  And do you need `fs.Stats`?  Using `fs.Stats` here limit this to Node only, and I don't see the stats method being used. You can use the `{}` typing if you don't care what it is.

Reply:
Done.